### PR TITLE
Add Alma linux 8 client support to libvirt

### PR DESCRIPTION
**Pull request by maximenoel8**
## What does this PR change?

This PR is created to add Alma Linux 8 to client for libvirt provider.

Currently, it has to be describe with the option `auto_register` set to `false`.

Packages to install spacewalk are missing in centos 8 repositories.

### CODE REVIEW

File: backend_modules/libvirt/host/user_data.yaml [hunk 1]
```diff
@@ -328,3 +328,40 @@ runcmd:
...
+  # repo for salt
+  tools_pool_repo:
+    baseurl: http://${ use_mirror_images ? mirror : "download.opensuse.org"}/repositories/systemsmanagement:/Uyuni:/Stable:/CentOS7-Uyuni-Client-Tools/CentOS_7/
+    failovermethod: priority
+    enabled: true
+    gpgcheck: false
+    name: tools_pool_repo
+    priority: 98
+  # repo for nss-mdns
+  salt-repo-almalinux:
```
Comment (rjmateus) [inline backend_modules/libvirt/host/user_data.yaml:343]: Do we need this repository to install which packages? I think salt should come from client tools

File: backend_modules/libvirt/host/user_data.yaml [hunk 2]
```diff
@@ -328,3 +328,40 @@ runcmd:
...
+    failovermethod: priority
+    enabled: true
+    gpgcheck: false
+    priority: 99
+    name: epel
+
+packages: ["avahi", "nss-mdns", "qemu-guest-agent"]
+
+runcmd:
+  #   HACK: cloud-init in Debian 10 does not take care of the following
```
Comment (rjmateus) [inline backend_modules/libvirt/host/user_data.yaml:362]: Is this needed or is some copy past leftovers from Debian 10?

File: backend_modules/libvirt/host/user_data.yaml [hunk 3]
```diff
@@ -328,3 +328,40 @@ runcmd:
...
+  - echo "PermitRootLogin yes" >> /etc/ssh/sshd_config
+  - systemctl restart sshd
+  - systemctl start qemu-guest-agent
+  - [ yum, install, -y, --nobest, salt-minion ]
```
Comment (rjmateus) [inline backend_modules/libvirt/host/user_data.yaml:366]: The salt package cannot be installed with the packages parameter?

File: backend_modules/libvirt/host/user_data.yaml [hunk 4]
```diff
@@ -328,3 +328,40 @@ runcmd:
 
 packages: ["salt-minion", "avahi-daemon", "qemu-guest-agent", "apt-transport-https", "python-apt", "python3-apt"]
 %{ endif }
+
+%{ if image == "almalinux8o" }
+yum_repos:
+  # repo for salt
+  tools_pool_repo:
+    baseurl: http://${ use_mirror_images ? mirror : "download.opensuse.org"}/repositories/systemsmanagement:/Uyuni:/Stable:/CentOS7-Uyuni-Client-Tools/CentOS_7/
```
Comment (rjmateus) [inline backend_modules/libvirt/host/user_data.yaml:336]: shouldn't `CentOS8` be used instead?

**Review by rjmateus — COMMENTED** [review]
Just some small comments. I didn't try to deploy it yet

File: backend_modules/libvirt/host/user_data.yaml (hunk 2 above)
Comment (maximenoel8) [reply backend_modules/libvirt/host/user_data.yaml:362]: Yes, they are needed, avahi for dhcp configuration and qemu-guest-agent for communication with host. It was failing without this 3 packages.

**Pull request merged**